src/lib/dnssd/Resolver_ImplMinimalMdns.cpp

```diff
@@ -277,19 +277,41 @@ class MinMdnsResolver : public Resolver, public MdnsPacketDelegate
     CHIP_ERROR Init(chip::Inet::EndPointManager<chip::Inet::UDPEndPoint> * udpEndPointManager) override;
     bool IsInitialized() override;
     void Shutdown() override;
-    void SetOperationalDelegate(OperationalResolveDelegate * delegate) override { mOperationalDelegate = delegate; }
-    void SetCommissioningDelegate(CommissioningResolveDelegate * delegate) override { mCommissioningDelegate = delegate; }
+    void SetOperationalDelegate(OperationalResolveDelegate * delegate) override
+    {
+        if (delegate == nullptr)
+        {
+            mOperationalDelegates.clear();
+        }
+        else
+        {
+            mOperationalDelegates.insert(delegate);
+        }
+    }
+    void SetCommissioningDelegate(CommissioningResolveDelegate * delegate) override
+    {
+        if (delegate == nullptr)
+        {
+            mCommissioningDelegates.clear();
+        }
+        else
+        {
+            mCommissioningDelegates.insert(delegate);
+        }
+    }
     CHIP_ERROR ResolveNodeId(const PeerId & peerId) override;
     void NodeIdResolutionNoLongerNeeded(const PeerId & peerId) override;
     CHIP_ERROR DiscoverCommissionableNodes(DiscoveryFilter filter = DiscoveryFilter()) override;
     CHIP_ERROR DiscoverCommissioners(DiscoveryFilter filter = DiscoveryFilter()) override;
+    CHIP_ERROR DiscoverOperational(DiscoveryFilter filter = DiscoveryFilter()) override;
+
     CHIP_ERROR StopDiscovery() override;
     CHIP_ERROR ReconfirmRecord(const char * hostname, Inet::IPAddress address, Inet::InterfaceId interfaceId) override;
 
 private:
-    OperationalResolveDelegate * mOperationalDelegate     = nullptr;
-    CommissioningResolveDelegate * mCommissioningDelegate = nullptr;
-    System::Layer * mSystemLayer                          = nullptr;
+    std::set<OperationalResolveDelegate *> mOperationalDelegates;
```

Comment:
We should avoid using STL in the core code for code size reasons. Why do you need multiple delegates?